src/sentry/seer/similarity/utils.py

```diff
@@ -179,18 +179,22 @@ def get_stacktrace_string(data: dict[str, Any]) -> str:
     html_frame_count = 0  # for a temporary metric
     stacktrace_str = ""
     found_non_snipped_context_line = False
+    is_frames_truncated = False
 
     metrics.distribution("seer.grouping.exceptions.length", len(exceptions))
 
     def _process_frames(frames: list[dict[str, Any]]) -> list[str]:
         nonlocal frame_count
         nonlocal html_frame_count
         nonlocal found_non_snipped_context_line
+        nonlocal is_frames_truncated
         frame_strings = []
 
         contributing_frames = [
             frame for frame in frames if frame.get("id") == "frame" and frame.get("contributes")
         ]
+        if len(contributing_frames) + frame_count > MAX_FRAME_COUNT:
+            is_frames_truncated = True
```

Comment:
It might just be unclear variable naming (and not having read through the entire algorithm) which is getting me, but why are we adding contributing frames to all frames?

Also, nit:

Comparisons return a bool, so we can just use that directly.

```suggestion
        is_frames_truncated =  len(contributing_frames) + frame_count > MAX_FRAME_COUNT
```

Reply:
we're doing this to see if the length of all the contributing frames we've accumulated so far (either from this current exception or some previous ones) is greater than 30

Reply:
Okay, so basially `contributing_frames` is ones we've already counted and `frame_count` is ones we're about to add?

Reply:
other way around, but yes that's the idea

Reply:
Got it, thanks for the explanation!